extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/health/ReactivePgDataSourceHealthCheck.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.reactive.pg.client.runtime.health;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import io.quarkus.arc.Arc;
+import io.vertx.pgclient.PgPool;
+
+@Readiness
+@ApplicationScoped
+public class ReactivePgDataSourceHealthCheck implements HealthCheck {
+    private PgPool pgPool;
+
+    @PostConstruct
+    protected void init() {
+        pgPool = Arc.container().instance(PgPool.class).get();
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Reactive PostgreSQL connection health check").up();
+
+        try {
+            CompletableFuture<Void> databaseConnectionAttempt = new CompletableFuture<>();
+            pgPool.query("SELECT 1", ar -> {
```

Comment:
Does "SELECT 1" trigger a roundtrip with the database? 
Just want to be sure.

Reply:
It should except if they do some weird magic in the driver but that would be surprising.